### ISSUE TITLE
Mempool should return error message for short contract names

### ISSUE THREAD

**Issue by pradel** [blockstack/stacks-blockchain#1703]
## Describe the bug

If you try to deploy a contract with a small name (eg: "test"), the node will reject the call with an http code 400 but do not contain any error message. This make it really hard to debug as you don't know what's the error is.

## Steps To Reproduce

Deploy the hello-world contract and name it "test".

## Expected behavior

You get an error message saying that the name is too short.

**Comment by diwakergupta** [comment]
@kantai to follow up with request for additional information.

**Comment by kantai** [comment]
Investigating this by first adding a unit test for mempool submissions with a short contract name -- @pradel, what did you use to create the contract publish? Can you provide any more detail on the exact commands you executed.

**Comment by pradel** [comment]
@kantai I created the transaction using `@blockstack/stacks-transactions`:

```ts
const network = new StacksTestnet();

const transaction = await makeContractDeploy({
          contractName: "test",
          codeBody: contract,
          senderKey: account.privateKeyHex,
          network,
          fee: new Big(values.fee),
        });

const data = await broadcastTransaction(transaction, network);
```

Then in the network tab, I could see that the server was not returning any error message if the name of the contract was lower than the minimum allowed.

**Comment by kantai** [comment]
Thanks @pradel! We'll look into this.

**Comment by diwakergupta** [comment]
Should be fixed by #1731
